lib/galaxy/tools/__init__.py

```diff
@@ -566,7 +585,7 @@ def remove_from_cache(self):
         source_path = self.tool_source._source_path
         if source_path:
             for region in self.app.toolbox.cache_regions.values():
-                region.delete(source_path)
+                region.pop(source_path, None)
```

Comment:
I think we need to call `persist_cache_region()` here.

Reply:
Yep, thanks! I've moved that into the caller so that we just need to persist once per set of removed tools.